<?xml version="1.0"?>
<launch>


<!-- get the arguments from the user -->
<arg name="traj" doc="Configuration profile to use" />
<arg name="debug" default="False" doc="Whether debug is on" />
<arg name="speed_factor" default="1.0" doc="the speed to run the trajectory at" />
<arg name="reps" default="1" doc="The number of reps" />
<arg name="replan" default="false" doc="decide whether to re-plan the trajectories on the fly" />
<arg name="start" default="0" doc="Where to start" />

<arg name="arm" default="true" doc="Use the arm" />
<arg name="hand" default="true" doc="Use the hand" />
<arg name="save" default="true" doc="Save data" />
<arg name="fake" default="false" doc="Save data" />
<arg name="use_camera" default="true" doc="Camera topic to save to" />

<arg name="id" default="" doc="Some output ID that will be appended to the data filename" />
<arg name="config" default="false" doc="Config file" />




<!-- start the run node and pass it all of the parameters -->
    <node name="pick_and_place_node" pkg="hand_arm" type="pick_place_run_multi.py" respawn="false"
    	output="screen">
      	<param name="traj_profile" type="str" value="$(arg traj)" />

		<param name="speed_factor" type="double" value="$(arg speed_factor)" />
		<param name="num_reps" type="int" value="$(arg reps)" />
		<param name="fake" type="bool" value="$(arg fake)" />
		<param name="out_id" type="str" value="$(arg id)" />

		<param name="replan" type="bool" value="$(arg replan)" unless="$(eval arg('config') !=0)" />
		<param name="start" type="int" value="$(arg start)" unless="$(eval arg('config') !=0)" />
		<param name="use_arm" type="bool" value="$(arg arm)" unless="$(eval arg('config') !=0)" />
		<param name="use_hand" type="bool" value="$(arg hand)" unless="$(eval arg('config') !=0)" />
		<param name="save_data" type="bool" value="$(arg save)" unless="$(eval arg('config') !=0)" />
		<param name="use_camera" type="bool" value="$(arg use_camera)" unless="$(eval arg('config') !=0)" />

		<rosparam command="load" file="$(find hand_arm)/config/run_configs/$(arg config)" if="$(eval arg('config') !=0)" />
    </node>
</launch>